file_store/src/file_info_poller.rs

```diff
@@ -54,51 +56,86 @@ pub struct FileInfoPoller<T> {
     p: PhantomData<T>,
 }
 
-impl<T> FileInfoPoller<T>
+#[derive(Debug, Clone)]
+pub struct FileInfoPollerServer<T> {
+    config: FileInfoPollerConfig<T>,
+    sender: Sender<FileInfoStream<T>>,
+}
+
+impl<T> FileInfoPollerConfigBuilder<T>
+where
+    T: Clone,
+{
+    pub fn create(self) -> Result<(Receiver<FileInfoStream<T>>, FileInfoPollerServer<T>)> {
+        let config = self.build().unwrap(); // TODO: fix the unwrap
```

Comment:
does this TODO need fixed before the initial release or are we cool merging this now and finishing later?

Reply:
fixed here: 0fab396f15e3ef14da478b36dec63ae7f32441f1